libr/io/map.c

```diff
@@ -305,6 +339,26 @@ R_API RIOMap* r_io_map_get(RIO* io, ut64 addr) {
 	return NULL;
 }
 
+R_API bool r_io_map_is_mapped(RIO* io, ut64 addr) {
+	if (!io) {
+		return false;
+	}
+#define CMP(addr, itv) \
+	(((addr) < r_itv_end (*(RInterval *)(itv))) ? -1 : (((addr) > r_itv_end (*(RInterval *)(itv))) ? 1 : 0))
```

Comment:
is the cast necessary? isnt *itv enough?

Reply:
yup, that seems necessary because the compiler complains:
```
map.c:357:42: error: passing 'void' to parameter of incompatible type 'RInterval' (aka 'struct r_interval_t')
        r_pvector_lower_bound (shadow, addr, i, CMP_END_GT);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
/Users/ftamagni/src/mrmacete-radare2/libr/include/r_vector.h:230:9: note: expanded from macro 'r_pvector_lower_bound'
                        if ((cmp (x, ((void **)(vec)->v.a)[m])) > 0) { \
                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
map.c:15:24: note: expanded from macro 'CMP_END_GT'
        (((addr) < r_itv_end (*(itv)) ? -1 : (((addr) > r_itv_end (*(itv))) ? 1 : 0))
                              ^~~~~~
/Users/ftamagni/src/mrmacete-radare2/libr/include/r_util/r_itv.h:38:40: note: passing argument to parameter 'itv' here
static inline ut64 r_itv_end(RInterval itv) {
```